src/main/java/ymfc/parser/Parser.java

```diff
@@ -515,6 +523,14 @@ public static void validateStepNumbers(List<String> stepStrings) throws InvalidA
                         + "\nLearn to count!");
             }
         }
+
+        // Verify the steps are in strict sequential order
+        for (int i = 0; i < stepNumbers.size(); i++) {
+            if (stepNumbers.get(i) != i + 1) {  // Steps should be exactly 1, 2, 3, ..., n
+                throw new InvalidArgumentException("Steps are not in order. Arrange them sequentially " +
+                        "(s1, s2, s3, ...)!");
+            }
+        }
```

Comment:
Will the user/tester complain about this? Will it directly affect the "fast-typing" CLI users? This is a indeed a nice solution, but would it cause some controversy among users/testers? 🤔

Reply:
technically the steps should be in order. If they aren't, it isn't about fast typing but they have a wrong input